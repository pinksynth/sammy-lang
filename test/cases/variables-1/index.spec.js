const {
  expectCompiledEval,
  performSnapshotAssertions,
} = require("../../helpers")
performSnapshotAssertions("Variables", __dirname)

test("Variables implementation", () => {
  expectCompiledEval(`
  a = 2
  b = 3

  a * b
  `).toBe(6)

  expectCompiledEval(() => `a + b`).toThrow(
    "Variable a on line 1 is not defined in the current scope."
  )
})
