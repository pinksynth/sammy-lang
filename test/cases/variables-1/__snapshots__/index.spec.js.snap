// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“¸ Variables 1`] = `
[
  {
    "columnNumberEnd": 11,
    "columnNumberStart": 1,
    "lineNumberEnd": 1,
    "lineNumberStart": 1,
    "tokenType": "VAR",
    "value": "first_name",
  },
  {
    "columnNumberEnd": 12,
    "columnNumberStart": 11,
    "lineNumberEnd": 1,
    "lineNumberStart": 1,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 13,
    "columnNumberStart": 12,
    "lineNumberEnd": 1,
    "lineNumberStart": 1,
    "tokenType": "ASSIGNMENT",
    "value": "=",
  },
  {
    "columnNumberEnd": 14,
    "columnNumberStart": 13,
    "lineNumberEnd": 1,
    "lineNumberStart": 1,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 23,
    "columnNumberStart": 14,
    "lineNumberEnd": 1,
    "lineNumberStart": 1,
    "tokenType": "STRING_PURE",
    "value": ""Lucille"",
  },
  {
    "columnNumberEnd": 1,
    "columnNumberStart": 23,
    "lineNumberEnd": 3,
    "lineNumberStart": 1,
    "tokenType": "WHITESPACE",
    "value": "

",
  },
  {
    "columnNumberEnd": 5,
    "columnNumberStart": 1,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "WEAK",
    "value": "weak",
  },
  {
    "columnNumberEnd": 6,
    "columnNumberStart": 5,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 15,
    "columnNumberStart": 6,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "VAR",
    "value": "last_name",
  },
  {
    "columnNumberEnd": 16,
    "columnNumberStart": 15,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 17,
    "columnNumberStart": 16,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "ASSIGNMENT",
    "value": "=",
  },
  {
    "columnNumberEnd": 18,
    "columnNumberStart": 17,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 25,
    "columnNumberStart": 18,
    "lineNumberEnd": 3,
    "lineNumberStart": 3,
    "tokenType": "STRING_PURE",
    "value": ""Bluth"",
  },
  {
    "columnNumberEnd": 1,
    "columnNumberStart": 25,
    "lineNumberEnd": 4,
    "lineNumberStart": 3,
    "tokenType": "WHITESPACE",
    "value": "
",
  },
  {
    "columnNumberEnd": 10,
    "columnNumberStart": 1,
    "lineNumberEnd": 4,
    "lineNumberStart": 4,
    "tokenType": "VAR",
    "value": "last_name",
  },
  {
    "columnNumberEnd": 11,
    "columnNumberStart": 10,
    "lineNumberEnd": 4,
    "lineNumberStart": 4,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 12,
    "columnNumberStart": 11,
    "lineNumberEnd": 4,
    "lineNumberStart": 4,
    "tokenType": "ASSIGNMENT",
    "value": "=",
  },
  {
    "columnNumberEnd": 13,
    "columnNumberStart": 12,
    "lineNumberEnd": 4,
    "lineNumberStart": 4,
    "tokenType": "WHITESPACE",
    "value": " ",
  },
  {
    "columnNumberEnd": 22,
    "columnNumberStart": 13,
    "lineNumberEnd": 4,
    "lineNumberStart": 4,
    "tokenType": "STRING_PURE",
    "value": ""Austero"",
  },
  {
    "columnNumberEnd": 1,
    "columnNumberStart": 22,
    "lineNumberEnd": 5,
    "lineNumberStart": 4,
    "tokenType": "WHITESPACE",
    "value": "
",
  },
]
`;

exports[`ðŸ“¸ Variables 2`] = `
{
  "children": [
    {
      "children": [
        {
          "columnNumberStart": 14,
          "interpolations": [],
          "lineNumberStart": 1,
          "subStrings": [
            "Lucille",
          ],
          "type": "LITERAL_STRING",
        },
      ],
      "columnNumberStart": 1,
      "lineNumberStart": 1,
      "type": "ASSIGNMENT",
      "variable": "first_name",
      "weak": false,
    },
    {
      "children": [
        {
          "columnNumberStart": 18,
          "interpolations": [],
          "lineNumberStart": 3,
          "subStrings": [
            "Bluth",
          ],
          "type": "LITERAL_STRING",
        },
      ],
      "columnNumberStart": 1,
      "lineNumberStart": 3,
      "type": "ASSIGNMENT",
      "variable": "last_name",
      "weak": true,
    },
    {
      "children": [
        {
          "columnNumberStart": 13,
          "interpolations": [],
          "lineNumberStart": 4,
          "subStrings": [
            "Austero",
          ],
          "type": "LITERAL_STRING",
        },
      ],
      "columnNumberStart": 1,
      "lineNumberStart": 4,
      "type": "ASSIGNMENT",
      "variable": "last_name",
      "weak": false,
    },
  ],
  "type": "ROOT",
}
`;

exports[`ðŸ“¸ Variables 3`] = `"const first_name=\`Lucille\`;let last_name=\`Bluth\`;last_name=\`Austero\`"`;
